logstash-core/src/main/java/org/logstash/health/Status.java

```diff
@@ -0,0 +1,32 @@
+package org.logstash.health;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum Status {
+    UNKNOWN,
+    GREEN,
+    YELLOW,
+    RED,
+    ;
+
+    private final String externalValue = name().toLowerCase();
+
+    @JsonValue
+    public String externalValue() {
+        return externalValue;
+    }
+
+    /**
+     * Combine this status with another status.
+     * This method is commutative.
+     * @param status the other status
+     * @return the more-degraded of the two statuses.
+     */
+    public Status reduce(Status status) {
```

Comment:
I couldn't get that _why_ we need reduce here?!

Reply:
It's part of future work, and slightly out of scope but a required part of getting this endpoint bootstrapped.

Once we have many indicator reports, we will need to reduce their individual statuses into a summary status. Similar with an indicator report from many probes, each of which can degrade status.